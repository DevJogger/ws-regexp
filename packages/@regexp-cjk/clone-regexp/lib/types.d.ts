import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';
export interface ICloneRegexp<T extends RegExp = RegExp> {
    (inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
}
export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags {
}
export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore {
    /**
     * allow change cloneRegexp function
     */
    cloneRegexp?: ICloneRegexp<T>;
    disableDetectRegexpClone?: boolean;
}
export declare type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
    clone?(...argv: any[]): any;
} ? T : T & {
    clone?(options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
};
