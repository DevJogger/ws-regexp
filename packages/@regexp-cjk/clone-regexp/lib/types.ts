import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';

export interface ICloneRegexp<T extends RegExp = RegExp>
{
	(inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv): T
}

export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags
{

}

export interface ICloneRegexpOptionsCustom<T extends RegExp = RegExp>
{
	/**
	 * allow change cloneRegexp function
	 */
	cloneRegexp?: ICloneRegexp<T>,

	disableDetectRegexpClone?: boolean,

	resetLastIndex?: boolean,

	lastIndex?: number,
}

export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore, ICloneRegexpOptionsCustom<T>
{

}

export type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
	clone?(...argv)
} ? T : T & {
	clone?(options?: ICloneRegexpOptionsCore, ...argv): T
}
