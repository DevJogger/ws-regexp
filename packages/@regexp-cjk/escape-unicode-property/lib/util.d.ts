import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
