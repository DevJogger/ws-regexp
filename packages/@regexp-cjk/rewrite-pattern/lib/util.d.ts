/**
 * Created by user on 2020/6/8.
 */
import { IOptionsRewritePattern } from '../index';
export declare function rewritePatternOptions(options?: IOptionsRewritePattern, flags?: string): IOptionsRewritePattern;
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
