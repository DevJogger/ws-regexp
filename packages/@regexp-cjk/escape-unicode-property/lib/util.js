"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const util_1 = require("@regexp-cjk/rewrite-pattern/lib/util");
const lodash_1 = require("lodash");
function handleOptions(options, flags) {
    return (0, util_1.handleOptions)((0, lodash_1.merge)({}, options, {
        unicodePropertyEscape: true,
        rewriteFlags: {
            unicode: true,
        },
    }), flags !== null && flags !== void 0 ? flags : 'u');
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map