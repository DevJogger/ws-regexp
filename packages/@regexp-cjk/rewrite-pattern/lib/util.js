"use strict";
/**
 * Created by user on 2020/6/8.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = exports.rewritePatternOptions = void 0;
const lodash_1 = require("lodash");
const rewrite_flags_1 = require("@regexp-cjk/rewrite-flags");
function rewritePatternOptions(options, flags) {
    var _a;
    return (0, lodash_1.defaults)({}, options, {
        dotAllFlag: true,
        lookbehind: true,
        namedGroup: true,
        useUnicodeFlag: (_a = flags === null || flags === void 0 ? void 0 : flags.includes) === null || _a === void 0 ? void 0 : _a.call(flags, 'u'),
    });
}
exports.rewritePatternOptions = rewritePatternOptions;
function handleOptions(options, flags) {
    var _a;
    flags = (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : flags;
    if (options === null || options === void 0 ? void 0 : options.rewriteFlags) {
        flags = (0, rewrite_flags_1.rewriteFlags)(flags, options.rewriteFlags);
    }
    options = rewritePatternOptions(options, flags);
    delete options.flags;
    return {
        options,
        flags,
    };
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map