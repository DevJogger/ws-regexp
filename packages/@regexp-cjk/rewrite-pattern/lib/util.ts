/**
 * Created by user on 2020/6/8.
 */

import { defaults } from 'lodash';
import { IOptionsRewritePattern } from '../index';
import { rewriteFlags } from '../flags';

export function rewritePatternOptions(options?: IOptionsRewritePattern, flags?: string): IOptionsRewritePattern
{
	return defaults<IOptionsRewritePattern, IOptionsRewritePattern, IOptionsRewritePattern>({}, options, {
		dotAllFlag: true,
		lookbehind: true,
		namedGroup: true,
		useUnicodeFlag: flags?.includes?.('u'),
	})
}

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	flags = options?.flags ?? flags;

	if (options?.rewriteFlags)
	{
		flags = rewriteFlags(flags, options.rewriteFlags);
	}

	options = rewritePatternOptions(options, flags)

	return {
		options,
		flags,
	}
}
