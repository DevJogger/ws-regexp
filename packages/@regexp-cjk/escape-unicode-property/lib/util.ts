import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions as _handleOptions } from '@regexp-cjk/rewrite-pattern/lib/util';
import { merge } from 'lodash';

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	return _handleOptions(merge({}, options, {
		unicodePropertyEscape: true,
		rewriteFlags: {
			unicode: true,
		},
	}), flags ?? 'u');
}
