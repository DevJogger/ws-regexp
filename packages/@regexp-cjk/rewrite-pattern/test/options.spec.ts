import { IOptionsRewritePattern } from '../index';
import { inspect } from "util";
import { handleOptions } from '../lib/util';
import { EnumFlagMap } from '../flags';

describe(`handleOptions`, () =>
{

	([
		{},
		{
			flags: "",
		},
		{
			options: {
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			flags: "u",
			options: {
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			options: {
				unicodePropertyEscape: false,
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			flags: "us",
			options: {
				rewriteFlags: {}
			},
		},
		{
			flags: "us",
			options: {
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			flags: "us",
			options: {
				rewriteFlags: {
					unicode: true,
				}
			},
		},
		{
			flags: "us",
			options: {
				flags: "usgi",
				rewriteFlags: {
					unicode: true,
				}
			},
			expected: {
				options: expect.anything(),
				flags: "usgi",
			}
		},
		{
			flags: "us",
			options: {
				flags: "usgi",
				rewriteFlags: {
					unicode: false,
				}
			},
			expected: {
				options: expect.anything(),
				flags: "sgi",
			}
		},
	] as {
		options?: IOptionsRewritePattern,
		flags?: string,
		expected?: {
			options?: IOptionsRewritePattern,
			flags?: string,
		}
	}[])
		.forEach(({ options, flags, expected }) =>
		{
			test(inspect({ options, flags }), () =>
			{

				let actual = handleOptions(options, flags);

				if (expected)
				{
					expect(actual).toStrictEqual(expected);
				}

				expect(actual.options).not.toHaveProperty('flags');

				expect(actual).toHaveProperty('flags');

				expect(actual.options)
					.toHaveProperty('dotAllFlag')
				;

				expect(actual.options)
					.toHaveProperty('lookbehind')
				;

				expect(actual.options)
					.toHaveProperty('namedGroup')
				;

				expect(actual.options)
					.toHaveProperty('useUnicodeFlag')
				;

				expect(actual).toMatchSnapshot();
			})

		})

});
