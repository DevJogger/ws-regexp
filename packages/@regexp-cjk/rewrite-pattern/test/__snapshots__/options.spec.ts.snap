// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleOptions {
  options: { flags: 'usgi', rewriteFlags: { unicode: false } },
  flags: 'us'
} 1`] = `
{
  "flags": "sgi",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": false,
    },
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions {
  options: { flags: 'usgi', rewriteFlags: { unicode: true } },
  flags: 'us'
} 1`] = `
{
  "flags": "usgi",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": true,
    },
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions {
  options: { unicodePropertyEscape: false, rewriteFlags: { unicode: false } },
  flags: undefined
} 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": false,
    },
    "unicodePropertyEscape": false,
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: 'u' } 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": false,
    },
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: 'us' } 1`] = `
{
  "flags": "s",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": false,
    },
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: undefined } 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": false,
    },
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: true } }, flags: 'us' } 1`] = `
{
  "flags": "us",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {
      "unicode": true,
    },
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: {} }, flags: 'us' } 1`] = `
{
  "flags": "us",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": {},
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: undefined, flags: '' } 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`handleOptions { options: undefined, flags: undefined } 1`] = `
{
  "flags": undefined,
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": undefined,
  },
}
`;
